### PR TITLE
Formatting and Style changes

### PR DESCRIPTION
**Pull request by SudheendraKatte**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?
Made changes to the tutorial from tech writing perspective (language, style, and format). Please review and let me know if the changes are okay.

### CODE REVIEW

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 1]
```diff
@@ -25,66 +37,90 @@ catch(Exception e) {
...
-Java exception tracking tooling can be used as a standalone capability or better yet, included as part of your CI/CD pipelines. Whenever code is exercised, you want to identify every exception and collected supporting details needed to fix them. This process can start as early as unit testing during CI and can continue through every other step of the software delivery lifecycle including in production.
+### Prevent using Java Exception Tracker
 
-Standalone exception tracking solutions can make it quicker to initially set up and get started tracking down and fixing exceptions. CI/CD integrated exception tracking solutions can take a little more time for initial setup but have greater potential for becoming a standard part of the software delivery process and adding more value to the organization overall.
+You need to identify the exceptions every time the code is checked-in and collect the logs required for fixing the exceptions. However, static analysis and testing do not detect all the issues. You need to perform runtime code analysis that identifies unknown issues and prevents them from being deployed. You can achieve this by using a Java Exception Tracker.
```
Comment (KimberlyFields) [inline tutorials/manage-service-reliability/intro-java-exception-management.md:61]:
```suggestion
You need to identify the exceptions every time the code is checked in and collect the logs required to fix the exceptions. However, static analysis and testing do not detect all the issues. You can use a Java Exception Tracker to perform runtime code analysis that identifies unknown issues and prevents them from being deployed. 
```

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 2]
```diff
@@ -25,66 +37,90 @@ catch(Exception e) {
...
 
-## Key Features of Java Exception Trackers
-What are the important features that a Java exception tracking solution should have? Here's a short list.
 
 ### Code Fingerprinting
-When the software analyzes code loaded into your JVM at runtime and assigns a unique fingerprint to each line of code. At runtime, exception tracking software should correlate each error and slowdown to its unique signature to identify anomalies and capture their complete state.
 
-### Source Code & Variables
-By analyzing and indexing the structure of the code, it's possible to collect the most relevant variable state information from running JVM whenever critical events take place. Data should be captured across the entire call stack up to 10 levels into heap.
+The Java Exception Tracker should have the capability to analyze the code loaded into your Java Virtual Machine (JVM) during runtime and assign a unique fingerprint to each line in the code. At runtime, the tracker should correlate each error, drilldown to its unique signature, identify anomalies, and capture the complete state.
```
Comment (KimberlyFields) [inline tutorials/manage-service-reliability/intro-java-exception-management.md:75]:
```suggestion
The Java Exception Tracker should have the ability to analyze the code loaded into your Java Virtual Machine (JVM) during runtime and assign a unique fingerprint to each line in the code. At runtime, the tracker should correlate each error, drill down to its unique signature, identify anomalies, and capture the complete state.
```

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 3]
```diff
@@ -25,66 +37,90 @@ catch(Exception e) {
...
+
+
+### Source Code and Variables
+
+Tracker should have the ability to analyze and index the code structure. This helps you collect data on the most relevant variable state from the running JVM when an event occurs. The tracker should capture data across the call stack up to 10 levels into heap.
```
Comment (KimberlyFields) [inline tutorials/manage-service-reliability/intro-java-exception-management.md:80]:
```suggestion
The tracker should have the ability to analyze and index the code structure. This helps you collect data on the most relevant variable state from the running JVM when an event occurs. The tracker should capture data across the call stack up to 10 levels into heap.
```

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 4]
```diff
@@ -25,66 +37,90 @@ catch(Exception e) {
...
 
+The tracker should have highly efficient cyclical in-memory buffers to store log statements and display the DEBUG/TRACE state of your code during the critical events, without increasing log verbosity or parsing log files from disk.
+
+ 
 ### Error Detection & Prioritization
-Exception tracking software should correlate all code events at runtime to their unique fingerprint (even as code changes over time) to identify and prioritize new and spiking errors without relying on developer foresight to log and search for them.
 
-### Host / Container State
-Capture detailed system metrics (e.g. CPU, GC) of the host / container in which the code is executing, as well as complete snapshots of the JVM state (e.g. env vars, threading state) at the moment of critical errors and slowdowns. You might need this data to fix the exception.
+The Java Exception Tracker should have the capability to correlate all the code events at runtime to their unique fingerprint (even as code changes over time). It should identify and prioritize new and spiking errors without relying on the developer’s foresight to log and search them.
```
Comment (KimberlyFields) [inline tutorials/manage-service-reliability/intro-java-exception-management.md:90]:
```suggestion
The Java Exception Tracker should have the capability to correlate all the code events at runtime to their unique fingerprint, even as code changes over time. It should identify and prioritize new and spiking errors without relying on the developer’s foresight to log and search them.
```

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 5]
```diff
@@ -25,66 +37,90 @@ catch(Exception e) {
...
+
+
+### Host/Container State
+
+The tracker should capture detailed system metrics (such as CPU, GC, and so on) of the host or container in which the code is executing. It should also capture complete snapshots of the JVM state such as env vars and threading state in the event of critical errors and slowdown.
```
Comment (KimberlyFields) [inline tutorials/manage-service-reliability/intro-java-exception-management.md:95]:
```suggestion
The tracker should capture detailed system metrics, such as CPU, GC, and so on, of the host or container in which the code is executing. It should also capture complete snapshots of the JVM state, such as env vars and threading state, in the event of critical errors and slowdown.
```

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 6]
```diff
@@ -25,66 +37,90 @@ catch(Exception e) {
...
-### PII (Personally Identifiable Information) Redaction
-Secure all sensitive data at all times! Redact PII variable data before it leaves the user environment. Make sure variable state is redacted based on configurable variable value patterns and code symbology: variable, field and class names.
+The tracker should have the capability to privately encrypt all the source code and the collected variable states using a 256-bit AES Encryption key before they leave the production node. 
+
 
+### Personally Identifiable Information (PII) Redaction​
 
-## Install the Harness Error Tracking Java Agent
-There are multiple options for installing the Harness Java Exception Tracking Agent. Click the link below and follow the documentation for your preferred type of installation.
+You must secure sensitive data at all times. The Java Exception Tracker should provide you the ability to redact the PII variable data before it leaves the user environment. You should ensure that the variable state is redacted based on the configurable variable value patterns and code symbology such as variable, field, and class names.
```
Comment (KimberlyFields) [inline tutorials/manage-service-reliability/intro-java-exception-management.md:105]:
```suggestion
You must secure sensitive data at all times. The Java Exception Tracker should provide you the ability to redact the PII variable data before it leaves the user environment. You should ensure that the variable state is redacted based on the configurable variable value patterns and code symbology, such as variable, field, and class names.
```

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 7]
```diff
@@ -25,66 +37,90 @@ catch(Exception e) {
...
 
-[Java Agent Installation Documentation](https://docs.harness.io/article/nx99xfcoxz-install-the-error-tracking-agent)
 
-## 4-Step Interactive Walkthrough
+## Demo - IRP Java Exceptions
 
-The walkthrough below is the quickest way to see what Harness has to offer when it comes to Java exception tracking. There is not a self guided free trial today but that is in the works. For now you can [request a demo](https://www.harness.io/interest/error-tracking) and someone from Harness will contact you to show you the solution and get you set up with the software.
+Here is a demo that shows you how to IRP Java exceptions using Harness Error Tracking Java Agent.
```
Comment (KimberlyFields) [inline tutorials/manage-service-reliability/intro-java-exception-management.md:110]:
```suggestion
Here is a demo that shows you how to IRP Java exceptions using the Harness Error Tracking Java Agent.
```

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 8]
```diff
@@ -25,66 +37,90 @@ catch(Exception e) {
...
+
+<!-- Video:
+https://harness-1.wistia.com/medias/rpv5vwzpxz-->
+<docvideo src="https://harness-1.wistia.com/medias/i6nqucdlv0" />
+
+
+## Harness Error Tracking Java Agent
+
+Here is an interactive video that demonstrates the capabilities of Harness Error Tracking Java Agent. Click on the glowing circle to see the capabilities. To know more about Harness Error Tracking Java Agent and request for a detailed demo, contact [Harness](https://www.harness.io/interest/error-tracking).
```
Comment (KimberlyFields) [inline tutorials/manage-service-reliability/intro-java-exception-management.md:119]:
```suggestion
Here is an interactive video that demonstrates the capabilities of the Harness Error Tracking Java Agent. Select the glowing circle to see the capabilities. To learn more about the Harness Error Tracking Java Agent and request a detailed demo, contact [Harness](https://www.harness.io/interest/error-tracking).
```

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 9]
```diff
@@ -25,66 +37,90 @@ catch(Exception e) {
...
 
 <div style={{ position: 'relative', paddingBottom: 'calc(52.1875% + 40px)', height: '0' }}><iframe src="https://demo.arcade.software/Lsj2sDDVCzMbQFnP7kvT?embed" frameborder="0" loading="lazy" webkitallowfullscreen mozallowfullscreen allowfullscreen style={{ position: 'absolute', top: '0', left: '0', width: '100%', height: '100%' }}></iframe></div>
 
+
+## Install Harness Error Tracking Java Agent
+
+You can install Harness Java Exception Tracking Agent in multiple ways. For more information, go to [Java Agent Installation Documentation](https://docs.harness.io/article/nx99xfcoxz-install-the-error-tracking-agent).
```
Comment (KimberlyFields) [inline tutorials/manage-service-reliability/intro-java-exception-management.md:126]:
```suggestion
You can install the Harness Java Exception Tracking Agent in multiple ways. For more information, go to [Java Agent Installation Documentation](https://docs.harness.io/article/nx99xfcoxz-install-the-error-tracking-agent).
```

**Review by KimberlyFields — CHANGES_REQUESTED** [review]

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 10]
```diff
@@ -5,16 +5,28 @@ keywords: [java exceptions, uncaught, swallowed exception, java errors, error tr
 ---
 
 # Finding and Fixing Java Exceptions
```
[the remark this answers is not in the document]
Comment (SudheendraKatte) [reply tutorials/manage-service-reliability/intro-java-exception-management.md:7]: Headings are now in sentence case.

Exceptions:

- Prevent using a Java Exception Tracker
- Key features of a Java Exception Tracker
- Personally Identifiable Information (PII) redaction​
- Harness Error Tracking Java Agent
- Install Harness Error Tracking Java Agent

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 11]
```diff
@@ -5,16 +5,28 @@ keywords: [java exceptions, uncaught, swallowed exception, java errors, error tr
...
-_**Caught exceptions**_ are handled in code via the `try` and `catch` keywords.
+According to [Oracle Java Documentation](https://docs.oracle.com/javase/tutorial/essential/exceptions/definition.html), *"An exception is an event, which occurs during the execution of a program, that disrupts the normal flow of the program's instructions."*
+
+
+## Types of Exceptions
+
+Types of Java exceptions are uncaught exceptions, caught exceptions, and swallowed exceptions.
+
+
+### Uncaught Exceptions
```
[the remark this answers is not in the document]
Comment (SudheendraKatte) [reply tutorials/manage-service-reliability/intro-java-exception-management.md:22]: Tired inserting a table but the code block is spoiling the formatting. Switched to bulleted list with headings in bold.

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 12]
```diff
@@ -4,87 +4,116 @@ description: Introducing Error Tracking - how to identify all exceptions and fix
...
+- **Swallowed exceptions** - Exceptions that are caught in empty `catch` blocks. They don't show up in logs. They are difficult to identify and troubleshoot, resulting in production incidents.
 
 
-## 2 Minute Demo Using Harness to IRP Java Exceptions
-<!-- Video:
-https://harness-1.wistia.com/medias/rpv5vwzpxz-->
-<docvideo src="https://harness-1.wistia.com/medias/i6nqucdlv0" />
+## Dealing with exceptions – a repeatable process
+
+Even though it is difficult to remove all the exceptions, it is important to identify the problematic exceptions and prevent them from escaping into production as they can negatively impact the users. Here is a three step Identify, Resolve, and Prevent (IRP) process that can help you minimize the bad exceptions.
```
Comment (KimberlyFields) [inline tutorials/manage-service-reliability/intro-java-exception-management.md:41]:
```suggestion
Even though it is difficult to remove all the exceptions, it is important to identify the problematic exceptions and prevent them from escaping into production as they can negatively impact the users. You can use a three-step process to identify, resolve, and prevent to minimize the bad exceptions.
```

**Review by ravilach — APPROVED** [review]
From the tech side, looks good to me. -Ravi

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 13]
```diff
@@ -4,87 +4,116 @@ description: Introducing Error Tracking - how to identify all exceptions and fix
...
+- **Swallowed exceptions** - Exceptions that are caught in empty `catch` blocks. They don't show up in logs. They are difficult to identify and troubleshoot, resulting in production incidents.
 
 
-## 2 Minute Demo Using Harness to IRP Java Exceptions
-<!-- Video:
-https://harness-1.wistia.com/medias/rpv5vwzpxz-->
-<docvideo src="https://harness-1.wistia.com/medias/i6nqucdlv0" />
+## Dealing with exceptions – a repeatable process
+
+Even though it is difficult to remove all the exceptions, it is important to identify the problematic exceptions and prevent them from escaping into production as they can negatively impact the users. You can use a three-step process to identify, resolve, and prevent to minimize the bad exceptions.
```
Comment (KimberlyFields) [inline tutorials/manage-service-reliability/intro-java-exception-management.md:41]:
```suggestion
Even though it is difficult to remove all the exceptions, it is important to identify the problematic exceptions and prevent them from escaping into production as they can negatively impact the users. You can use a three-step process to identify, resolve, and prevent bad exceptions.
```

File: tutorials/manage-service-reliability/intro-java-exception-management.md [hunk 14]
```diff
@@ -4,87 +4,116 @@ description: Introducing Error Tracking - how to identify all exceptions and fix
...
+
+### Identify critical issues
+
+Proactively identify runtime exceptions and slowdowns in every release including issues that swallowed exceptions. It's important to know when the new releases introduce new, increasing, or resurfaced exceptions.
+
+### Resolve using complete context
+
+Reproduce any exception or slowdown with the complete source code, variables, DEBUG logs, and environment state. You need purpose-built tools such as Harness Java Exception Tracker to do this. After you get the context, fix all the exceptions, even if they are not logged.
+
+### Prevent using a Java Exception Tracker
```
Comment (KimberlyFields) [inline tutorials/manage-service-reliability/intro-java-exception-management.md:52]:
```suggestion
### Prevent exceptions using a Java Exception Tracker
```

**Review by KimberlyFields — APPROVED** [review]
I added two more comments to fix things I missed last time. Please correct those items. You can merge the PR after you fix them.

**Pull request merged**